### PR TITLE
media upload clear button

### PR DESCRIPTION
**Pull request by Robert-Costello**
## Product Description

This adds a clear button to media upload questions that clears the input and makes a call to formplayer, which then deletes the file from the file system. 
clear-button

## Technical Summary

jira ticket: https://dimagi-dev.atlassian.net/browse/USH-3448
spec doc: https://docs.google.com/document/d/1lRYWe-IRUWN7OTYK9mVvGYhPoy_Wo85NnF0dooSw1rY/edit#heading=h.d7s2sycdfk92
formplayer PR: https://github.com/dimagi/formplayer/pull/1459

## Feature Flag

USH: Support signature, image, audio, and video questions in Web Apps
### Safety story

changes restricted to feature flag
tested locally and on staging

### QA Plan

no QA planned

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

**Review by shubham1g5 — COMMENTED** [review]
@Robert-Costello I am seeing a bug on staging with this (also visible in your screencapture above) where clicking on 'clear' doesn't clear the Green check mark.

**Comment by Robert-Costello** [comment]
> clicking on 'clear' doesn't clear the Green check mark.

Thanks, I'll take a look.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js [hunk 1]
```diff
@@ -892,6 +892,13 @@ hqDefine("cloudcare/js/form_entry/entries", function () {
         };
         self.file = ko.observable();
         self.extensionsMap = initialPageData.get("valid_multimedia_extensions_map");
+        self.onClear = function () {
+            self.file(null);
+            self.rawAnswer(constants.NO_ANSWER);
+            self.xformAction = constants.CLEAR_ANSWER;
+            self.question.onClear();
+            self.xformAction= constants.ANSWER_MEDIA;
```
[the remark this answers is not in the document]
Comment (Robert-Costello) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/entries.js:900]: That's a reasonable suspicion. I wasn't 100% confident with how `xformAction` is being handled here. I'm now thinking of updating it in the success and failure callbacks. In either case (success or failure) we'd want the `xformAction` to be set back to `constants.ANSWER_MEDIA`. That should eliminate any concerns around asynchronous issues.

File: corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js [hunk 2]
```diff
@@ -746,6 +746,14 @@ hqDefine("cloudcare/js/form_entry/form_ui", function () {
         }, self.throttle);
         self.onchange = self.triggerAnswer;
 
+        self.triggerClear = function () {
+            publishClearEvent();
+        };
+        var publishClearEvent = _.throttle(function () {
+            $.publish('formplayer.' + constants.CLEAR_ANSWER, self);
+        }, self.throttle)
+        self.onClear = self.triggerClear;
```
[the remark this answers is not in the document]
Comment (Robert-Costello) [reply corehq/apps/cloudcare/static/cloudcare/js/form_entry/form_ui.js:755]: That right, I was following the `triggerAnswer/publishAnswerEvent` pattern. It makes sene to simplify `onClear`.

**Review by MartinRiese — APPROVED** [review]

**Pull request merged**